File: onion-architecture/java/src/main/java/com/example/infrastructure/jpa/JpaStudentEntity.java
package com.example.infrastructure.jpa;

import java.util.*;

@Entity
public class JpaStudentEntity {

    @Id
    @GeneratedValue
    private Long id;
    private String name;

    @ManyToMany
    private Set<JpaCourseEntity> enrolledCourses = new HashSet<>();

    // getters/setters
}

File: onion-architecture/kotlin/src/main/kotlin/com/example/infrastructure/jpa/JpaStudentEntity.kt
package com.example.infrastructure.jpa

import jakarta.persistence.*

@Entity
class JpaStudentEntity {

    @Id
    @GeneratedValue
    var id: Long? = null
    
    var name: String? = null

    @ManyToMany
    var enrolledCourses: MutableSet<JpaCourseEntity> = mutableSetOf()

    // getters/setters handled by Kotlin properties
}